### DISCUSSION TITLE
Will the messages filtered by tags on the consumer side be marked as consumed? | 在consumer端被tag过滤掉的消息会被标记已消费吗？

### DISCUSSION THREAD

**Discussion by leopardrota** [apache/rocketmq#6878] [Q&A]
The message filtering based on tags in RocketMQ is performed twice, first by the broker and then by the consumer. I would like to ask if RocketMQ marks the messages filtered out at the consumer end as consumed. If it does not mark them, how does it achieve this, given that the messages have already been sent to the consumer end?

---
RocketMQ根据tag的消息过滤是在broker和consumer先后过滤了一遍。我想问，在consumer端被过滤掉的消息，RocketMQ是否标记该消息已被消费？如果不标记，是怎么做到的，毕竟消息都已经发到consumer端？

**Comment by fuyou001** [comment]
如果是tag订阅，会在服务端过滤。
判断有没有消费，就是根据消息的位点，如果同一个订阅者同一个queue，如果消息被tag过滤了，正好这个被过滤的后面一条消息被消费，那sdk会提交当前的消息位点。（包括了被tag过滤的 消息，以及tag之后的一条消息的位点）
